std/math/algebraic.d

```diff
@@ -369,6 +369,86 @@ real hypot(real x, real y) @safe pure nothrow @nogc
         }
 }
 
+/***********************************************************************
+ * Calculates the distance of the point (x, y, z) from the origin (0, 0, 0)
+ * in three-dimensional space.
+ * The distance is the value of the square root of the sums of the squares
+ * of x, y, and z:
+ *
+ *      sqrt($(POWER x, 2) + $(POWER y, 2) + $(POWER z, 2))
+ *
+ * Note that the distance between two points (x1, y1, z1) and (x2, y2, z2)
+ * in three-dimensional space can be calculated as hypot(x2-x1, y2-y1, z2-z1).
+ *
+ * Params:
+ *     x = floating point value
+ *     y = floating point value
+ *     z = floating point value
+ *
+ * Returns:
+ *     The square root of the sum of the squares of the given arguments.
+ */
+T hypot(T)(const T x, const T y, const T z) @safe pure nothrow @nogc
+if (isFloatingPoint!T)
+{
+    import std.math : fmax;
+    const absx = fabs(x);
+    const absy = fabs(y);
+    const absz = fabs(z);
+
+    // Scale all parameters to avoid overflow.
+    const ratio = fmax(absx, fmax(absy, absz));
+    if (ratio == 0.0)
+        return ratio;
+
+    return ratio * sqrt((absx / ratio) * (absx / ratio)
+                        + (absy / ratio) * (absy / ratio)
+                        + (absz / ratio) * (absz / ratio));
+}
+
+///
+@safe unittest
+{
+    import std.math : feqrel;
+
+    assert(hypot(1.0, 2.0, 2.0) == 3.0);
+    assert(hypot(2.0, 3.0, 6.0) == 7.0);
+    assert(hypot(1.0, 4.0, 8.0) == 9.0);
```

Comment:
IMHO, this should use `isClose` - and `feqrel` isn't used at all...

Reply:
Was a copy-pasto from the other hypot unittest block.